apps/metric_processors.go

```diff
@@ -42,7 +41,9 @@ func (p MetricsProcessorExcludeMetrics) Processors() []otel.Component {
 		for _, g := range strings.Split(glob, "*") {
 			literals = append(literals, regexp.QuoteMeta(g))
```

Comment:
I don't think the `QuoteMeta` is going to escape `$` properly (it's presumably going to do `\$` instead of `$$`)

Reply:
Actually I think maybe it needs to be `\$$`? So do a `strings.Replace` after `QuoteMeta`?

Reply:
Are you talking about the case where the customer puts a `$` into an `exclude_metrics` glob? Is `$` even a valid character in metric names [1]? Should we just add validation instead?

[1] https://cloud.google.com/monitoring/api/v3/naming-conventions#naming-types-and-labels

Reply:
Yeah, I'm talking about the case where `$` is in the glob. Even if it's not allowed by GCM, I think OT still allows it, and either way we shouldn't generate a bad config file. I'm happy with properly escaping it or forbidding it with a validator.

Reply:
Added validation and a test that specifically checks for a `$` in the glob.

Reply:
Just realized that this newly-added validation will eventually reject patterns for some of our third-party workload metrics (e.g., `workload.googleapis.com/jvm.memory.heap.*` — this pattern is not possible today because of the `endswith=/*` rule, but will become reasonable once that restriction is relaxed). Should we change the validation now, or wait until it actually becomes an issue (I assume we're not renaming the metrics at this point)?
@qingling128 FYI.

Reply:
Relaxing this validation would make it possible for users to exclude specific metrics. I think that's useful, but let's run it by Harp via internal channel for potential concerns he had expressed before.

Reply:
Yeah, the naming conventions doc referenced above seems inaccurate. I'm trying to confirm the actual validation, so we can determine whether `$` can appear in metric types. b/207141764 tracks the doc update.

Reply:
We are currently also enforcing that any metric patterns in `exclude_metrics` filters start with `agent.googleapis.com/`, which will also prevent excluding workload metrics. If we do relax the restrictions, we should relax both of those.

That said, I've removed the validation and added the code to escape the `$` instead.

Reply:
A-and it turns out we don't need to escape the `$` in filterprocessor.

Reply:
The trailing `$` happens to work when un-escaped, but we do need the escapes. Sigh.